src/praisonai/tests/unit/integrations/test_managed_agents.py

```
@@ -124,74 +131,95 @@ async def stream(self, prompt: str, **kwargs) -> AsyncIterator[Dict[str, Any]]:
+    
+    # Check that LocalManagedAgent implements required protocol methods
+    local_agent = LocalManagedAgent()
+    
+    # Test protocol compliance at runtime
+    assert hasattr(local_agent, 'execute')
+    assert hasattr(local_agent, 'stream') 
+    assert hasattr(local_agent, 'reset_session')
+    assert hasattr(local_agent, 'reset_all')
+    
+    # Test method signatures exist and are callable
+    assert callable(getattr(local_agent, 'execute'))
```

Ruff (0.15.10)

[warning] Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)